### PR TITLE
Warn users if they are using an old version

### LINKED ISSUE #340: Check if senza is updated
Senza should warn users if there's a new version available in pypi. This check should done only once a day to avoid wasting time and resources.

### PR DESCRIPTION
**Pull request by jmcs**
Fixes #340.

### CODE REVIEW

File: senza/subcommands/root.py [hunk 1]
```diff
@@ -1,16 +1,98 @@
...
+    """
+    current_version = LooseVersion(current_version)
+    latest_version = get_latest_version()
+
+    if current_version < latest_version:
+        if __file__.startswith('/home'):
+            # if it's installed in the user folder
+            cmd = "pip install --upgrade stups-senza"
+        else:
+            cmd = "sudo pip install --upgrade stups-senza"
```
Comment (rafaelcaricio) [inline senza/subcommands/root.py]: I think we should be careful on telling users which command they should use to upgrade Senza. We do not really know how they installed their Senza. Maybe we should simply say that there's a new version available.

**Comment by rafaelcaricio** [comment]
:+1:

File: senza/subcommands/root.py [hunk 2]
```diff
@@ -1,16 +1,98 @@
...
+                latest_version = LooseVersion(str_version)
+    return latest_version
+
+
+def get_latest_version_from_pypi() -> str:
+    """
+    Gets the latest release version pypi api using distutils order
+    to sort the releases (same as pip).
+    """
+    pypi_response = requests.get(PYPI_URL)
```
Comment (hjacobs) [inline senza/subcommands/root.py]: timeout? error handling?

**Comment by hjacobs** [comment]
I think we need to catch all errors in the version checking --- Senza should never fail or get unnecessary slow just because of PyPI downtimes or similar...

File: senza/subcommands/root.py (hunk 2 above)
Comment (jmcs) [inline senza/subcommands/root.py]: Done.

**Comment by rafaelcaricio** [comment]
:+1:

**Comment by jmcs** [comment]
@hjacobs can you re-review this PR. It now [ignores all the exceptions](https://github.com/zalando-stups/senza/pull/358/commits/7ba7b5843fa46aa3ab9151ec6f91fdc71e24debe#diff-cae6888e502721ef34399fe531c1bf28R87).

File: senza/subcommands/root.py [hunk 3]
```diff
@@ -1,16 +1,112 @@
...
 from ..arguments import GLOBAL_OPTIONS, region_option
+from ..error_handling import sentry
 
 CONTEXT_SETTINGS = dict(help_option_names=['-h', '--help'])
 
+PYPI_URL = "https://pypi.python.org/pypi/stups-senza/json"
+ONE_DAY = 86400  # seconds
+
+
+def get_latest_version_from_disc() -> Optional[LooseVersion]:
```
Comment (jmcs) [inline senza/subcommands/root.py]: Fixed.

**Comment by hjacobs** [comment]
@jmcs let's resolve the conflicts and merge :smile:

**Comment by jmcs** [comment]
@hjacobs also done.

**Comment by hjacobs** [comment]
:+1:

**Pull request merged**